src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -197,8 +197,11 @@ public static Quantity parseQuantity(String quantity) throws ParseException {
     public static Date parseDate(String date) throws ParseException {
         requireNonNull(date);
         String trimmedDate = date.trim();
-        if (!Date.isValidDate(trimmedDate)) {
-            throw new ParseException(Date.MESSAGE_CONSTRAINTS);
+        if (!Date.isValidDateFormat(trimmedDate)) {
+            throw new ParseException(Date.MESSAGE_CONSTRAINTS_WRONG_FORMAT);
+        }
+        if (!Date.isValidDateInput(trimmedDate)) {
+            throw new ParseException(Date.MESSAGE_CONSTRAINTS_WRONG_DATE);
```

Comment:
Both conditions in both if statements are the same, how would the 2nd one ever run?